File: MySQL-NET-REST-API/DocumentManagement.Data/Entities/StorageSetting.cs
using System;

namespace DocumentManagement.Data
{
    public class StorageSetting : BaseEntity
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string JsonValue { get; set; }
        public bool IsDefault { get; set; }
        public bool EnableEncryption { get; set; }
        public StorageType StorageType { get; set; }
    }
}

File: AngularFrontend/src/app/storage-setting/storage-setting.ts
import { StorageType } from "./storage-type-enum";

export interface StorageSetting<T> {
  id?: string;
  storageType: StorageType;
  name: string;
  isDefault: boolean;
  enableEncryption: boolean;
  jsonValue?: T;
  isEditing?: boolean;
}

export interface AWSS3storage {
  accessKey: string;
  secretKey: string;
  bucketName: string;
  region: string;
}

export interface CloudflareStorage {
  accessKey: string;
  accountID: string;
  secretKey: string;
  bucketName: string;
  region: string;
}
